static/app/components/tooltip.tsx

```diff
@@ -116,215 +111,262 @@ function computeOriginFromArrow(
   }
 }
 
-class Tooltip extends React.Component<Props, State> {
-  static defaultProps: DefaultProps = {
-    position: 'top',
-    containerDisplayMode: 'inline-block',
-  };
-
-  state: State = {
-    isOpen: false,
-    usesGlobalPortal: true,
-  };
-
-  async componentDidMount() {
-    if (IS_ACCEPTANCE_TEST) {
-      const TooltipStore = (await import('sentry/stores/tooltipStore')).default;
-      TooltipStore.addTooltip(this);
+function getTooltipPortal(useGlobalPortal: boolean): HTMLElement {
+  // Use a single global portal element for tooltips
+  if (useGlobalPortal) {
+    let portal = document.getElementById('tooltip-portal');
+    if (!portal) {
```

Comment:
Do we try to encourage early returns btw? You could 

```tsx
if(portal) {
  return portal
}

...create and return it
```

Reply:
I'm going to leave it like this _for now_ since there's a few other places we have similar code, will refactor / cleanup a bit later